### PR TITLE
fix error message for __call__ for non-callable objects

### LINKED ISSUE #1416: Wrong error message when calling a non-callable object
CPython

```
>>> 'linux'()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object is not callable
>>> 1()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not callable
```

RustPython

```
>>>>> 'linux'()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unsupported method: __call__
>>>>> 1()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Unsupported method: __call__
```

### PR DESCRIPTION
**Pull request by pitachips**
closes #1416.

Changed only `vm.rs` file, and the other files were just auto-formatted with `cargo fmt`

### CODE REVIEW

**Comment by coolreader18** [comment]
Could you upgrade your rust version to `1.38`? The reason that the other files were auto-formatted is because master is up to date with 1.38

**Comment by pitachips** [comment]
Upgrading fixed the problem. Thanks for letting me know.

**Comment by dralley** [comment]
Are there any test snippets for this?

File: vm/src/vm.rs [hunk 1]
```diff
@@ -563,7 +563,7 @@ impl VirtualMachine {
                 let wrapped = self.call_get_descriptor(func, obj.clone())?;
                 self.invoke(&wrapped, args)
             }
-            None => Err(self.new_type_error(format!("Unsupported method: {}", method_name))),
+            None => Err(self.new_type_error(format!("{:?} object is not callable", cls.name))),
```
Comment (coolreader18) [inline vm/src/vm.rs]: You should put the change in `vm.invoke` instead, maybe check if it's callable using `vm.is_callable` before calling `call_method`.

**Comment by pitachips** [comment]
> Are there any test snippets for this?

I added tests that check not just Error type but also message, since the previous code also raised `TypeError`. However I am not sure if this error message comparison is a good choice, because I am afraid Python may change error messages in future versions. Any suggestions?

**Comment by coolreader18** [comment]
I don't think any test snippets are necessary, as this is mainly just a UX change.

**Comment by pitachips** [comment]
Alright, I'll just get rid of the added tests.

**Pull request merged**